### PR TITLE
fix(VDatePicker): allow null value for multiple date picker

### LINKED ISSUE #9529: [Bug Report] Clear causing error when using v-date-picker with v-text-field
### Environment
**Vuetify Version:** 2.1.7
**Vue Version:** 2.6.10
**Browsers:** Chrome 78.0.3904.70
**OS:** Linux x86_64

### Steps to reproduce
Create a vDatePicker with range=true in menu using a clearable vTextField as explained in the docs. Select any date/range and the click the clear button in the vTextField 

### Expected Behavior
vTextField clear button should set the target v-model to an emtpy array [ ] to prevent array errors when target v-model is an array

### Actual Behavior
vTextField clear button sets the target v-model to null which causes the error. This works fine for a vDatePicker without range property because target v-model is not an array, nor does the vDatePicker itself expect an array

### Reproduction Link
<a href="https://codepen.io/hope_feda/pen/abbVdoj" target="_blank">https://codepen.io/hope_feda/pen/abbVdoj</a>

### Other comments
I tried setting the target v-model to an empty array [ ] using the click:clear event. However, this did not work because apparently click:clear doesn't override the default behaviour of setting the target v-model to null.
I'm not sure if the solution is that the clear event should set the target v-model to an empty array if its an array originally(null when not an array), or if click:clear event should override default behaviour so we can customize how its cleared.

### PR DESCRIPTION
**Pull request by jacekkarczmarczyk**
## Description
Allows null/undefined value for range/multiple date pickers

fix, but deliberately dev branch

## Motivation and Context
fixes #9529

## How Has This Been Tested?
jest, playground

## Markup:

<details>

```vue
<template>
  <v-menu
    :close-on-content-click="false"
    max-width="290"
  >
    <template #activator="{ on }">
      <v-text-field
        v-model="value"
        clearable
        :label="JSON.stringify(value)"
        readonly
        v-on="on"
      />
      <v-checkbox
        v-model="isRange"
        label="true = range, false = mutliple"
      />
    </template>
    <v-date-picker
      v-model="value"
      v-bind="{ [isRange ? 'range' : 'multiple']: true }"
    />
  </v-menu>
</template>

<script>
  export default {
    data: () => ({ value: null, isRange: true }),
  }
</script>
```
</details>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/vuetify/src/components/VDatePicker/VDatePicker.ts [hunk 1]
```diff
@@ -134,25 +135,30 @@ export default mixins(
           return this.pickerDate
         }
 
-        const date = (this.multiple || this.range ? (this.value as string[])[(this.value as string[]).length - 1] : this.value) ||
+        const multipleValue = this.value ? wrapInArray(this.value) : []
```
Comment (johnleider) [inline packages/vuetify/src/components/VDatePicker/VDatePicker.ts:138]: If we have access to `this` could we use `this.multipleValue`?

File: packages/vuetify/src/components/VDatePicker/VDatePicker.ts [hunk 2]
```diff
@@ -250,15 +256,15 @@ export default mixins(
...
+      } else if (this.isMultiple && this.multipleValue.length && (!oldValue || !(oldValue as string[]).length) && !this.pickerDate) {
         this.tableDate = sanitizeDateString(this.inputDate, this.type === 'month' ? 'year' : 'month')
       }
     },
     type (type: DatePickerType) {
       this.activePicker = type.toUpperCase()
 
       if (this.value && this.value.length) {
-        const output = (this.isMultiple ? (this.value as string[]) : [this.value as string])
+        const output = wrapInArray(this.value)
```
Comment (johnleider) [inline packages/vuetify/src/components/VDatePicker/VDatePicker.ts:267]: Same here, could we use `multipleValue`?

**Comment by johnleider** [comment]
Should this target master now that 2.3 is out?

**Comment by jacekkarczmarczyk** [comment]
yes, I'll rebase

File: packages/vuetify/src/components/VDatePicker/VDatePicker.ts (hunk 1 above)
Comment (jacekkarczmarczyk) [reply packages/vuetify/src/components/VDatePicker/VDatePicker.ts]: computed accessed in data() returns undefined, `this.$options.computed.multipleValue.call(this)` works but it doesn't make sense

File: packages/vuetify/src/components/VDatePicker/VDatePicker.ts [hunk 3]
```diff
@@ -7,18 +7,19 @@ import VDatePickerYears from './VDatePickerYears'
 
 // Mixins
 import Localable from '../../mixins/localable'
-import mixins from '../../util/mixins'
 import Picker from '../../mixins/picker'
 
 // Utils
+import mixins from '../../util/mixins'
+import isDateAllowed from './util/isDateAllowed'
+import { wrapInArray } from '../../util/helpers'
```
Comment (johnleider) [inline packages/vuetify/src/components/VDatePicker/VDatePicker.ts:15]: order alphabetically

File: packages/vuetify/src/components/VDatePicker/VDatePicker.ts [hunk 4]
```diff
@@ -293,11 +298,11 @@ export default mixins(
...
   methods: {
     emitInput (newInput: string) {
-      if (this.range && this.value) {
-        if (this.value.length !== 1) {
+      if (this.range) {
+        if (this.multipleValue.length !== 1) {
           this.$emit('input', [newInput])
         } else {
-          const output = [...this.value, newInput]
+          const output = [this.multipleValue[0], newInput]
```
Comment (johnleider) [inline packages/vuetify/src/components/VDatePicker/VDatePicker.ts:305]: is it possible for **multipleValue** to be 0 and if so is **undefined** an acceptable value for **output** on 305?

Comment (jacekkarczmarczyk) [reply packages/vuetify/src/components/VDatePicker/VDatePicker.ts:305]: not possible, in this branch of `if` length of multipleValue is always equal 1

**Pull request merged**